tests/components/color_extractor/test_service.py

```diff
@@ -0,0 +1,319 @@
+"""Tests for color_extractor component service calls."""
+import base64
+import io
+
+import aiohttp
+import pytest
+from voluptuous.error import MultipleInvalid
+
+from homeassistant.components.color_extractor import (
+    ATTR_PATH,
+    ATTR_URL,
+    DOMAIN,
+    SERVICE_TURN_ON,
+)
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_BRIGHTNESS_PCT,
+    ATTR_RGB_COLOR,
+    DOMAIN as LIGHT_DOMAIN,
+    SERVICE_TURN_OFF as LIGHT_SERVICE_TURN_OFF,
+)
+from homeassistant.const import ATTR_ENTITY_ID, STATE_OFF, STATE_ON
+from homeassistant.setup import async_setup_component
+import homeassistant.util.color as color_util
+
+from tests.async_mock import Mock, mock_open, patch
+from tests.common import load_fixture
+
+LIGHT_ENTITY = "light.kitchen_lights"
+CLOSE_THRESHOLD = 10
+
+
+def _close_enough(actual_rgb, testing_rgb):
+    """Validate the given RGB value is in acceptable tolerance."""
+    # Convert the given RGB values to hue / saturation and then back again
+    # as it wasn't reading the same RGB value set against it.
+    actual_hs = color_util.color_RGB_to_hs(*actual_rgb)
+    actual_rgb = color_util.color_hs_to_RGB(*actual_hs)
+
+    testing_hs = color_util.color_RGB_to_hs(*testing_rgb)
+    testing_rgb = color_util.color_hs_to_RGB(*testing_hs)
+
+    actual_red, actual_green, actual_blue = actual_rgb
+    testing_red, testing_green, testing_blue = testing_rgb
+
+    r_diff = abs(actual_red - testing_red)
+    g_diff = abs(actual_green - testing_green)
+    b_diff = abs(actual_blue - testing_blue)
+
+    return (
+        r_diff <= CLOSE_THRESHOLD
+        and g_diff <= CLOSE_THRESHOLD
+        and b_diff <= CLOSE_THRESHOLD
+    )
+
+
+@pytest.fixture(autouse=True)
+async def setup_light(hass):
+    """Configure our light component to work against for testing."""
+    assert await async_setup_component(
+        hass, LIGHT_DOMAIN, {LIGHT_DOMAIN: {"platform": "demo"}}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(LIGHT_ENTITY)
+    assert state
+
+    # Validate starting values
+    assert state.state == STATE_ON
+    assert state.attributes.get(ATTR_BRIGHTNESS) == 180
+    assert state.attributes.get(ATTR_RGB_COLOR) == (255, 63, 111)
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        LIGHT_SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: LIGHT_ENTITY},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(LIGHT_ENTITY)
+
+    assert state
+    assert state.state == STATE_OFF
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)